From c1ab4470357feeac524d0b83827026ba549cb3b0 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Tue, 27 Oct 2020 18:01:44 +0100
Subject: GCC: move attribute to the start of function definition in VAAPI
 wrapper.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix compilation error:
../../base/compiler_specific.h:97:28: error: attributes are not allowed on a function-definition
   97 | #define WARN_UNUSED_RESULT __attribute__((warn_unused_result))
      |                            ^~~~~~~~~~~~~
../../media/gpu/vaapi/vaapi_wrapper.h:322:36: note: in expansion of macro ‘WARN_UNUSED_RESULT’
  322 |                     const T* data) WARN_UNUSED_RESULT {
      |                                    ^~~~~~~~~~~~~~~~~~

Bug: 819294
Change-Id: I24e2e10a96182f8705fed96164550357acdb55a4
---
 media/gpu/vaapi/vaapi_wrapper.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/gpu/vaapi/vaapi_wrapper.h b/media/gpu/vaapi/vaapi_wrapper.h
index fd1fd823ee1e..deeda1f0e4bc 100644
--- a/media/gpu/vaapi/vaapi_wrapper.h
+++ b/media/gpu/vaapi/vaapi_wrapper.h
@@ -318,8 +318,8 @@ class MEDIA_GPU_EXPORT VaapiWrapper
   // Convenient templatized version of SubmitBuffer() where |size| is deduced to
   // be the size of the type of |*data|.
   template <typename T>
-  bool SubmitBuffer(VABufferType va_buffer_type,
-                    const T* data) WARN_UNUSED_RESULT {
+  bool WARN_UNUSED_RESULT SubmitBuffer(VABufferType va_buffer_type,
+                                       const T* data) {
     return SubmitBuffer(va_buffer_type, sizeof(T), data);
   }
   // Batch-version of SubmitBuffer(), where the lock for accessing libva is
-- 
2.25.1

